lib/coinbase/staking_operation.rb

```diff
@@ -104,6 +110,30 @@ def status
       @model.status
     end
 
+    # Returns whether the Staking Operation is in a terminal state.
+    # @return [Boolean] Whether the Staking Operation is in a terminal state
+    def terminal_state?
+      failed_state? || complete_state?
+    end
+
+    # Returns whether the Staking Operation is in a failed state.
+    # @return [Boolean] Whether the Staking Operation is in a failed state
+    def failed_state?
+      status == 'failed'
+    end
+
+    # Returns whether the Staking Operation is in a complete state.
+    # @return [Boolean] Whether the Staking Operation is in a complete state
+    def complete_state?
```

Comment:
```suggestion
    def completed?
```

Reply:
@alex-stone do you want me to change `terminal_state?` also to `terminal?` ?

Reply:
The way these translate for readability is something like the following:

"is operation completed" => "operation.completed?"
"is operation failed" => "operation.failed?"
"is operation in a terminal state" => "operation.terminal_state?"

So I think `terminal_state?` is appropriate!

Reply:
Decided to keep terminal_state as is, as I see the same pattern in all other classes like Transaction, Transfer etc

Reply:
Done 👍